src/qt/anybutton.cpp

```diff
@@ -33,9 +34,74 @@ class wxQtPushButton : public wxQtEventSignalHandler< QPushButton, wxAnyButton >
 wxQtPushButton::wxQtPushButton(wxWindow *parent, wxAnyButton *handler)
     : wxQtEventSignalHandler< QPushButton, wxAnyButton >( parent, handler )
 {
+    installEventFilter( this );
     connect(this, &QPushButton::clicked, this, &wxQtPushButton::clicked);
 }
 
+bool wxQtPushButton::eventFilter(QObject *watched, QEvent *event)
+{
+    if( watched != this )
+        return false;
+    QPushButton *button = qobject_cast<QPushButton *>( watched );
+    wxAnyButton *wxbutton = GetHandler();
+    QIcon icon = button->icon();
+    if( event->type() == QEvent::HoverMove && button->isEnabled() )
+    {
+        if( wxbutton->GetStateBitmaps()[1].IsOk() )
```

Comment:
This code cries for a helper function checking if a bitmap with the given index is valid and setting it if it is. Having such helper would make this function twice shorter and ten times more clear.